default callback - add "show_task_path_on_failure" option

samdoran opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
When running in verbosity <2, display the file and line number for tasks that fail. This provides
useful information without having to run at increased verbosity.

Fixes #64625

- [x] Add tests
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request


##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
`lib/ansible/plugins/callback/default.py`

samdoran committed:
default callback - add "show_task_path_on_failure" option

When running in verbosity <2, display the file and line number for tasks that fail. This provides
useful information without having to run at increased verbosity.

Co-authored-by: tahar.jegham <jeghamseifeddine@gmail.com>

webknjaz commented on changelogs/fragments/64625-show-file-path-on-task-failure-callback-option.yml:
```diff
@@ -0,0 +1,5 @@
+  - >
+    default callback - add ``show_task_path_on_failure`` option to display file and
+    line number of tasks only on failed tasks when running at
+    normal verbosity lev (https://github.com/ansible/ansible/issues/64625)
```
```suggestion
    normal verbosity level (https://github.com/ansible/ansible/issues/64625)
```
?

webknjaz commented on changelogs/fragments/64625-show-file-path-on-task-failure-callback-option.yml:
```diff
@@ -0,0 +1,5 @@
+minor_changes:
+  - >
```
Did you mean
```suggestion
  - >-
```

samdoran committed:
Fixup changelog

samdoran committed:
Move _print_task_path to CallbackBase class

samdoran committed:
Remove incorrect path printing code

Not sure why/how this snuck in.

samdoran committed:
Add integration tests

samdoran committed:
Fix indentation in docs.

bcoca commented on lib/ansible/plugins/callback/__init__.py:
```diff
@@ -259,6 +259,11 @@ def _clean_results(self, result, task_name):
+    def _print_task_path(self, task):
+        path = task.get_path()
+        if path:
+            self._display.display(u"task path: %s" % path, color=C.COLOR_DEBUG)
```
you probalby want same color as origin, default to debug

samdoran replied:
Should that be `C.COLOR_ERROR` or `None` when verbosity < 2 called from `v2_runner_on_failed()`? I think it would be odd to have it be different colors depending on verbosity level, which is why I just kept it as `C.COLOR_DEBUG`.

bcoca replied:
well, you might be displaying it in diff levels, so probably that should be the configuable part: `(self, task, display=v|error|warning)` or just return the text to the caller, let the caller actually display?

samdoran committed:
Add color parameter to _print_task_path()

samdoran commented:
/azp run

samdoran committed:
Keep color output consistent for now

Currently the path is display with COLOR_DEBUG formatting with verbosity >= 2.
Instead of the color of the path changing based on verbosity level, just keep it at the
currenty behavior of COLOR_DEBUG. Having the color of the same information change
based on verbosity level seems incorrect and makes visual parsing of the information
more difficult.
